commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-1.16.4/src/main/java/dev/jorel/commandapi/nms/ExceptionHandlingArgumentSerializer_1_16_4_R3.java

```diff
@@ -0,0 +1,88 @@
+package dev.jorel.commandapi.nms;
+
+import com.google.gson.JsonObject;
+import com.mojang.brigadier.arguments.ArgumentType;
+import dev.jorel.commandapi.CommandAPIHandler;
+import dev.jorel.commandapi.arguments.ExceptionHandlingArgumentType;
+import net.minecraft.server.v1_16_R3.ArgumentRegistry;
+import net.minecraft.server.v1_16_R3.ArgumentSerializer;
+import net.minecraft.server.v1_16_R3.PacketDataSerializer;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+
+public class ExceptionHandlingArgumentSerializer_1_16_4_R3<T> implements ArgumentSerializer<ExceptionHandlingArgumentType<T>> {
+
+	private static Method getInfo = null;
+
+	@Override
+	public void a(ExceptionHandlingArgumentType<T> argument, PacketDataSerializer packetDataSerializer) {
```

Comment:
I don't think I can? Since I'm overriding obfuscated methods I have to use the same obfuscated names. I suppose I could do something like:
```java
@Override
public void a (parameters) {
    serialize(parameters);
}

public void serialize(parameters) {
    // Implementation...
}
```
Should I do this instead for each method?

Reply:
Ah, my bad - didn't realize you were implementing from a version-specific NMS class `ArgumentSerializer`. Leave the obfuscated method names, but include a comment stating a reasonable name instead, such as:

```java
@Override
// serializeToNetwork
public void a(parameters) {
    // ...
}

@Override
// serializeToJson
public void a(parameters) {
    // ...
}

@Override
// deserializeFromNetwork
public void b(parameters) {
    // ...
}

```